### PR TITLE
Adding lockpicks.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Adds a set of lockpicks with high lockpicking effectiveness.

## Why and what will this PR improve
Currently the best lockpicking tool in the codebase is a stick or a metal rod.

## Authorship
Code by myself, base icon by Zirc.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Review by out-of-phaze — COMMENTED** [review]
huh i kinda figured these would be a storage object. doesn't this mean the whole thing will just disintegrate when you fail to pick a lock?

File: code/game/objects/items/lockpicks.dm [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	name                = "roll of lockpicks"
+	desc                = "A stitched roll used to store thin, strangely-shaped tools commonly used used to pick locks."
+	icon                = 'icons/obj/items/lockpick_roll.dmi'
+	icon_state          = ICON_STATE_WORLD
+	material            = /decl/material/solid/organic/leather
+	color               = /decl/material/solid/organic/leather::color
+	material_alteration = MAT_FLAG_ALTERATION_ALL
+	storage             = /datum/storage/lockpick_roll
+
+/obj/item/lockpick_roll/filled/Initialize(ml, material_key)
```
Comment (out-of-phaze) [inline code/game/objects/items/lockpicks.dm]: this should be WillContain, shouldn't it?

**Review by out-of-phaze — CHANGES_REQUESTED** [review]

File: code/game/objects/items/lockpicks.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/items/lockpicks.dm]: Oh true.

File: code/game/objects/items/lockpicks.dm [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/obj/item/lockpick/hook
+	use_icon_state = "2"
+/obj/item/lockpick/lever
+	use_icon_state = "3"
+
+/obj/item/lockpick/Initialize(ml, material_key)
+	. = ..()
+	if(isnull(use_icon_state))
+		use_icon_state = num2text(rand(1,3))
+		icon_state = "[get_world_inventory_state()][use_icon_state]"
```
Comment (out-of-phaze) [inline code/game/objects/items/lockpicks.dm]: is there a reason this isn't just an `update_icon()` call?

File: code/game/objects/items/lockpicks.dm [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	. = ..()
+	icon_state = ICON_STATE_WORLD
+	if(!storage?.opened)
+		icon_state = "[icon_state]-rolled"
+	else if(length(contents))
+		var/x_offset = -9
+		for(var/i = 1 to length(contents))
+			var/obj/item/lockpick/thing = contents[i]
+			if(istype(thing))
+				var/image/pick         = new /image
```
Comment (out-of-phaze) [inline code/game/objects/items/lockpicks.dm]: might be better to just do an appearance copy?
```dm
				var/image/pick         = new /image
				pick.appearance = thing
				pick.appearance_flags |= RESET_COLOR
				pick.pixel_x = x_offset
```
also pick is a global proc (builtin, technically), not super certain about using it as a keyword :(

**Review by out-of-phaze — CHANGES_REQUESTED** [review]

File: code/game/objects/items/lockpicks.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/game/objects/items/lockpicks.dm]: I did an appearance copy in the previous revision of this and it broke colouration and layering for some reason.

File: code/game/objects/items/lockpicks.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/objects/items/lockpicks.dm]: There is a reason but I don't remember what it is.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**